src/Schema/AbstractAsset.php

```diff
@@ -52,7 +52,7 @@ protected function _setName($name)
         }
 
         if (strpos($name, '.') !== false) {
-            $parts            = explode('.', $name);
+            $parts            = explode('.', $name, 2);
```

Comment:
@OliverSkroblin, thank you for the patch. Please see the discussion in https://github.com/doctrine/dbal/pull/4708/files#r695330660.

Reply:
Hi @morozov,

I have read through the discussion and understand you also fundamentally. Nevertheless, I am of the opinion that the change should be accepted because it is wrong from a logical point of view.

In the whole schema manager, the full name of the index is used. Only when creating the detailed table information, this name is exploded and put back together incompletely. 

You further discuss, in the linked thread, that this has to be supported differently depending on the database platform. What I don't understand is that the database platform provides this name, which is then not processed uniformly in the abstraction level.

The name of the index, is different in the different layers of the Schema Manager, which nevertheless leads to incorrect assumptions being made.

Reply:
Seems that I am not the only one having issues with that 😖 

@OliverSkroblin you are making a good point here. When a db vendor did not support dots it is no change and when they do, it always had the option to break when db designers placed more than one dot. This `explode` was doomed to leave data behind.

Reply:
I support this change as well, without 2, the remaining part is skipped which is very dangerous.

However, there are more unsafe explode usage in this project and all of them should be fixed.

Reply:
> the remaining part is skipped which is very dangerous.
Which renaming part do you mean?

> there are more unsafe explode usage in this project and all of them should be fixed.
This may be possible. But these are not related to the explode in the AbstractAsset class, because the AbstractAsset class only explodes the name, which is then used for further processing. Other explode are not related to this change in my opinion.

Reply:
Well, the first step would be to stop removing the dots from the names of the different elements. I am currently working very intensively with the Schema Manager. 

**Background:** It should be possible for us to extend the schema of our product via plugins using XML files. For this I use the Schema Manager to rewrite foreign keys and tables.

The patch that is made here, I have already added via composer path manipulation, and so far no further problems with dots in indices have occurred.

The schema manager also accepts the names in all other places.

Reply:
> If this patch is accepted and we also want to add the support for dots in the schema name, how do we go about it?

Does this mean we might be missing string delimiters or name escaping symbols like ` or " at some places? What thoughts do you have in mind on "dots in schema name" support?

Reply:
> Does this mean we might be missing string delimiters or name escaping symbols like ` or " at some places?

Not sure how this is related to the support of the dot in object names but yes. See examples starting 
https://github.com/doctrine/dbal/issues/4357#issuecomment-911153565.

> What thoughts do you have in mind on "dots in schema name" support?

I believe it should be supported as well as any other valid characters in the names of all types of objects. See https://github.com/doctrine/dbal/issues/4772.